### PR TITLE
Pl dev

### PR DESCRIPTION
**Pull request by paulflang**
- improves handling of logdir
- adds plotting functionality
- uses most recent API of `readSBML`

### CODE REVIEW

**Comment by anandijain** [comment]
looks like you have conflicts still maybe you can squash and clean up this PR a bit?

File: src/suite.jl [hunk 1]
```diff
@@ -38,19 +38,33 @@ function verify_case(dir;verbose=false)
...
+        rs = ReactionSystem(ml)
         sys = ODESystem(ml)
+        statenames = [string(s.f.name) for s in sys.states]
+        
         ts = LinRange(settings["start"], settings["duration"], settings["steps"])
         prob = ODEProblem(sys, Pair[], (settings["start"], Float64(settings["duration"])); saveat=ts)
-        sol = solve(prob, CVODE_BDF(); abstol=settings["absolute"], reltol=settings["relative"])
-        solm = Array(sol)'
-        m = Matrix(results[1:end-1, 2:end])
+        sol = solve(prob, CVODE_BDF(); abstol=settings["absolute"]/20, reltol=settings["relative"]/20)
```
Comment (anandijain) [inline src/suite.jl:55]: why divide by 20?

File: src/suite.jl [hunk 2]
```diff
@@ -38,19 +38,33 @@ function verify_case(dir;verbose=false)
...
+        solm = hcat(sol.u...)'
```
Comment (anandijain) [inline src/suite.jl:56]: why?

File: src/suite.jl [hunk 3]
```diff
@@ -38,19 +38,33 @@ function verify_case(dir;verbose=false)
...
+        m = Matrix(results[1:end-1, 2:end])[:, sortperm(statenames)]
```
Comment (anandijain) [inline src/suite.jl:57]: are you saying they were just in the wrong order?

File: src/suite.jl [hunk 4]
```diff
@@ -38,19 +38,33 @@ function verify_case(dir;verbose=false)
...
         res = isapprox(solm, m; atol=1e-2)
+        if !isapprox(solm, m; atol=1e-9, rtol=3e-2)
```
Comment (anandijain) [inline src/suite.jl:59]: i'd add a kwarg that defaults to false so that we don't default to saving plots

File: src/suite.jl [hunk 5]
```diff
@@ -62,7 +76,7 @@ end
 
 function verify_all(;verbose=true)
     df = DataFrame(dir=String[], retcode=Bool[], atol=Float64[], error=String[])
-    ds = filter(isdir, readdir(joinpath(@__DIR__, "../data/sbml-test-suite/semantic/"); join=true))
+    ds = filter(isdir, readdir(joinpath(datadir, "sbml-test-suite", "semantic"); join=true))[55:150]
```
Comment (anandijain) [inline src/suite.jl:79]: remove the [55:150]

File: test/test_suite.jl [hunk 6]
```diff
@@ -1,35 +1,47 @@
...
 # @test sum(length.([good, bad])) == 1664
 
 now_fmtd = Dates.format(now(), dateformat"yyyy-mm-dd\THH-MM-SS")
-suite_df = lower_fns(suite_fns; write_fn="test_suite_$(now_fmtd).csv", verbose=true)
+# suite_df = lower_fns(suite_fns; write_fn="test_suite_$(now_fmtd).csv", verbose=true)
 # suite_df = lower_fns_threaded(suite_fns; write_folder="logs/suite/", write_fn="test_suite_$(now_fmtd).csv", verbose=true)
-@show suite_df
-@info nrow(filter(suite_df, :retcode => x -> x == 5)) "num good ones"
+# @show suite_df
+# @info nrow(filter(suite_df, :retcode => x -> x == 5)) "num good ones"
```
Comment (anandijain) [inline test/test_suite.jl:27]: add these back in (24, 26,27)

File: src/suite.jl (hunk 3 above)
Comment (paulflang) [reply src/suite.jl:57]: yes. although surprisingly many of them were in the right order. But we actually need to `sortperm(sortperm(statenemes))`

File: src/suite.jl (hunk 1 above)
Comment (paulflang) [reply src/suite.jl]: sorry. i just tried sth. fixed in a2e2dce .

File: src/suite.jl (hunk 2 above)
Comment (paulflang) [reply src/suite.jl]: thought that might fix the order but it didn't. fixed in a2e2dce .

File: src/suite.jl (hunk 4 above)
Comment (paulflang) [reply src/suite.jl]: i used the `verbose` kwarg in a2e2dce that was previously unused.

File: src/suite.jl (hunk 5 above)
Comment (paulflang) [reply src/suite.jl]: fixed in a2e2dce .

File: test/test_suite.jl [hunk 7]
```diff
@@ -1,35 +1,47 @@
...
+using SBML
+using CSV
+using Test
+using Dates
+
 sbml_test_suite()
 
 println("****SBML TEST SUITE TESTING****")
-suite_fns = get_sbml_suite_fns()
+suite_fns = get_sbml_suite_fns()[1:5]
```
[the remark this answers is not in the document]
Comment (paulflang) [reply test/test_suite.jl]: fixed in a2e2dce .

File: test/test_suite.jl (hunk 6 above)
Comment (paulflang) [reply test/test_suite.jl]: fixed in a2e2dce .

File: src/suite.jl [hunk 8]
```diff
@@ -1,89 +1,89 @@
...
-    for dir in ds
-        ret = verify_case(dir; verbose=verbose)
-        verbose && @info ret 
-        push!(df, ret)
-    end
-    verbose && print(df)
-    df
-end
+"""
+requires git
```
Comment (anandijain) [inline src/suite.jl:2]: i think mirek was saying sth here. might be my fault, but why is the whole file replaced?

Comment (paulflang) [reply src/suite.jl:2]: no idea. looks pretty much the same though. maybe i messed sth up using `git rm --cached`? what shall we do?

Comment (anandijain) [reply src/suite.jl:2]: ok i dont care enough about this bs to have it hold us up lol

**Pull request merged**